test/models/test_cice.py

```diff
@@ -121,24 +151,202 @@ def test_setup(ice_history):
     # Check config files are moved to model's work path
     work_path_files = os.listdir(model.work_path)
     assert CICE_NML_NAME in work_path_files
+    assert HIST_NML_NAME not in work_path_files
 
     # Check cice_in was patched with ice_history
     work_input_fpath = os.path.join(model.work_path, CICE_NML_NAME)
     input_nml = f90nml.read(work_input_fpath)
-    if ice_history:
-        assert input_nml["icefields_nml"] == ice_history["icefields_nml"]
+    if cice_config_files['ice_history']:
+        assert (input_nml["icefields_nml"] ==
+                cice_config_files["ice_history"]["icefields_nml"])
     else:
         assert input_nml["icefields_nml"] == DEFAULT_CICE_NML["icefields_nml"]
 
     # Check dump_last doesn't exist
     with pytest.raises(KeyError, match="dump_last"):
         input_nml["setup_nml"]["dump_last"]
 
-    # cleanup
-    with cd(expt_workdir):
-        os.remove(cice_nml["setup_nml"]["pointer_file"])
-        os.rmdir(cice_nml["setup_nml"]["restart_dir"])
+
+def test_no_restart_ptr(cice_config_files):
```

Comment:
Maybe you are right this should be a seperate issue, because ideally we would cover all the variations in the comment below right? 

e.g.

prior restart path specified/ not specified
restart from nml is true/false
restart ptr missing/not missing

which is possible 6 permutations